pkg/metrics/metrics.go

```diff
@@ -0,0 +1,98 @@
+package metrics
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Error to be  returned when marketplace is unable to reach quay.io.
+type OpsrcError string
+
+const (
+	// Default values which are to be used as labels
+	// for prometheus metrics.
+	Error     = "opsrc_error"
+	OpsrcName = "opsrc_name"
+
+	// Path, Port and Host where custom metrics would be exposed.
+	MetricsPath = "/metrics"
+	MetricsPort = "8383"
+	MetricsHost = "0.0.0.0"
+
+	// Error message to be returned when quay.io is down.
+	UnreachableError OpsrcError = "quay_unavailabe"
+
+	// Error message to be returned when host address is wrong.
+	NoHostExsistsError OpsrcError = "no_host_exsists"
+)
+
+// Metric variables that report error based on the operator-source
+// which fails to reconcile.
+var (
+	operatorSourceError = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "operator_source_failure_count",
+			Help: "Monotonic count of times operator-source could not download packages",
+		},
+		[]string{OpsrcName, Error},
+	)
+	// List of custom metrics which are to be collected.
+	metricsList = []prometheus.Collector{
+		operatorSourceError,
+	}
+)
+
+// registerMetrics registers the metrics with prometheus.
+func registerMetrics() error {
+	for _, metric := range metricsList {
+		err := prometheus.Register(metric)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// RecordMetrics starts the server, records the custom metrics and exposes
+// the metrics at the specified endpoint.
+func RecordMetrics() {
+	// Register metrics for the operator with the prometheus.
+	registerMetrics()
+
+	// Start the server and expose the registered metrics.
+	http.Handle(MetricsPath, promhttp.Handler())
+	port := ":" + MetricsPort
```

Comment:
If MetricsPort was a string, you could set port to `fmt.Sprintf(":%d", MetricsPort)`.